packages/storybook-addon/src/lib/storybook-addon.ts

```diff
@@ -2,18 +2,15 @@ import fs from 'fs';
 import { dirname, join } from 'path';
 import * as process from 'process';
 import VirtualModulesPlugin from 'webpack-virtual-modules';
-import { container, Configuration } from 'webpack';
+import { container, type Configuration } from 'webpack';
 import { logger } from '@storybook/node-logger';
-// NOTE: @storybook/core-common is deprecated while still available, considering importing
-// from 'storybook/internal/common' or '@storybook/core'. Considering requires Storybook 8
-// at least and change this in the next breaking change version.
-import { normalizeStories } from '@storybook/core-common';
-import { ModuleFederationPluginOptions } from '@module-federation/utilities';
-
-import { ModuleFederationConfig } from '@nx/webpack';
+import { normalizeStories } from '@storybook/core/common';
 import withModuleFederation from '../utils/with-module-federation';
 import { correctImportPath } from '../utils/correctImportPath';
 
+import type { ModuleFederationPluginOptions } from '@module-federation/utilities';
```

Comment:
Don't use the utility package. It's dead. Source types from sdk etc.

Reply:
`module-federation/sdk` should have the options, otherwise look in enhanced plugin to see where we import its types from and use those